scripts/release-helm.sh

```diff
@@ -50,6 +50,17 @@ scripts/helm-dependencies.sh validate
 # Read the shared scripting config.
 source "distribution/helm/scripting-config.sh"
 
+# Filter out vector and aggregator if not publishing to nightly
+FILTER_ONE="vector"
+FILTER_TWO="vector-aggregator"
+if [ "${CHANNEL}" != "nightly" ]; then
+    for i in "${!CHARTS_TO_PUBLISH[@]}"; do
+        if [[ "${CHARTS_TO_PUBLISH[$i]}" == "$FILTER_ONE" || "${CHARTS_TO_PUBLISH[$i]}" == "$FILTER_TWO" ]]; then
+            unset "CHARTS_TO_PUBLISH[$i]"
+        fi
+    done
+fi
+
```

Comment:
```suggestion
# Filter out vector and aggregator if not publishing to nightly
CHARTS_TO_SKIP_ON_NON_NIGHTLY=("vector" "vector-aggregator")
if [[ "${CHANNEL}" != "nightly" ]]; then
    for IDX in "${!CHARTS_TO_PUBLISH[@]}"; do
        if [[ "${CHARTS_TO_SKIP_ON_NON_NIGHTLY[@]}" =~ "${CHARTS_TO_PUBLISH[$IDX]}" ]]; then
            unset "CHARTS_TO_PUBLISH[$IDX]"
        fi
    done
fi

```

A couple of changes:
1. Use of `[[` built-in rather than external command `[`
2. `IDX` instead of `i` for better variable naming.
3. Joined the filter values into an array - this way we can add more pretty easily if we need to.